### PR TITLE
Dump stack traces from all lthreads on SIGTERM

### PR DESCRIPTION
**Pull request by prp**
This PR makes the lthread scheduler display stack traces for all lthreads after the launcher has received a SIGTERM. This means that CI tests that time out will now display stack traces because the timeout command sends a SIGTERM.

It works by installing a SIGTERM signal handler in the launcher and using a new ecall to output the stack traces. 

This is only supported in DEBUG builds.

In addition the PR also:
- fixes a merge conflict related to the outputting of mount tables;
- fixes the SGXLKL_TRACE_SIGNAL environment variable.

This is to help get more information when issue https://github.com/lsds/sgx-lkl/issues/748 occurs in the CI, as I find it hard to reproduce it locally.

### CODE REVIEW

File: src/lkl/setup.c [hunk 1]
```diff
@@ -1243,6 +1239,10 @@ static void* lkl_termination_thread(void* args)
                 "Could not unmount disk %d, %s\n", i, lkl_strerror(res));
         }
 
+#ifdef DEBUG
+    display_mount_table();
```
Comment (davidchisnall) [inline src/lkl/setup.c]: Why is this moved?  Don't we want the mount table from before we unmount everything?

File: src/main-oe/sgxlkl_run_oe.c [hunk 2]
```diff
@@ -331,14 +335,20 @@ static void help_enclave_config()
...
-/* Control the event channel notification between host & guest */
-
-/* Signal handler to resume paused evt chn to resume guest */
+#ifdef DEBUG
+// Signal handler for thread stack dumps and virtio event channel tests
 static void sgxlkl_loader_signal_handler(int signo)
 {
     switch (signo)
     {
+        case SIGTERM:
```
Comment (davidchisnall) [inline src/main-oe/sgxlkl_run_oe.c:344]: Can we do this for SIGUSR1 as well?  It would be convenient to have a way of doing this form outside without stopping the program.

File: src/sgxlkl.edl [hunk 3]
```diff
@@ -31,6 +31,11 @@ enclave {
         // Enclave call for initializing ethreads to enter enclave
         public void sgxlkl_ethread_init(void);
 
+        // Enclave call to dump stack traces for all lthreads (DEBUG only)
+        // TODO: This should only be included for a DEBUG build, but EDL doesn't seem to
```
Comment (davidchisnall) [inline src/sgxlkl.edl:35]: It does have include.  We could have a separate top-level EDL file for debug and non-debug builds and include the non-debug one in the debug build.

**Review by davidchisnall — APPROVED** [review]

File: src/lkl/setup.c (hunk 1 above)
Comment (prp) [reply src/lkl/setup.c]: Before the wrong merge, the idea was to output the mount table twice, before the unmount and after, to check that things have been correctly unmounted. The broken merge meant that we are outputting the mount table twice before the unmount, which  outputs the same information twice.

As an aside, I am not convinced that the unmouting always reliably flushes the file system state back to disk. I may open an issue on this.

File: src/main-oe/sgxlkl_run_oe.c (hunk 2 above)
Comment (prp) [reply src/main-oe/sgxlkl_run_oe.c:344]: Ok, that's an easy change to make (especially since the stack traces always come from the scheduler state and do not interrupt thread execution).

File: src/sgxlkl.edl (hunk 3 above)
Comment (prp) [reply src/sgxlkl.edl:35]: Ah, ok. I suggest that this should be a separate PR though, as it's related to #144 and #171.

Comment (davidchisnall) [reply src/sgxlkl.edl:35]: No objections.

File: src/lkl/setup.c (hunk 1 above)
Comment (prp) [reply src/lkl/setup.c]: Ok, I didn't move it to the correct location, but I fixed this now.

I've also opened this issue: https://github.com/lsds/sgx-lkl/issues/754
